### PR TITLE
Fixed: Beaming section

### LINKED ISSUE #45: Clarify beam nesting
Currently the specification does not say anything about disallowing beam nesting.

### LINKED ISSUE #84: Clarify beam groups in Mensural notation
Currently thinking that this should be disallowed.

### PR DESCRIPTION
**Pull request by ahankinson**
- Disallows nested beams
 - Clarified allowed values within beams
 - Disallows beam groups in mensural notation

Fixes #45
Fixes #84
Refs #24

### CODE REVIEW

File: v2/index.html [hunk 1]
```diff
@@ -1160,32 +1160,36 @@ <h4>Chords</h4>
...
-            </table>
-
+            <div class="issue" data-number="84"></div>
+            <p>
+                Beamed notes are encoded using braces <code>{}</code>.
+            </p>
+            <p>
+                Beam groups MUST start with an opening brace <code>{</code>, and
+                MUST end with a closing brace, <code>}</code>. Nested beam groups
+                MUST NOT occur.
```
Comment (lpugin) [inline v2/index.html:1172]: Except in a grace group `'8{CDqq{GF}rEF}`
image

File: v2/index.html [hunk 2]
```diff
@@ -1160,32 +1160,36 @@ <h4>Chords</h4>
...
+            </p>
+            <p>
+                Notes with duration values of <code>0</code>, <code>1</code>, <code>2</code>,
+                <code>4</code>, and <code>9</code> MUST NOT occur within
+                a beam group. Notes with other duration values MAY occur within
+                a beam group.
+            </p>
+            <p>
+                Rests SHOULD NOT occur within a beam group.
```
Comment (lpugin) [inline v2/index.html:1181]: I don't think we can say that.

File: v2/index.html [hunk 3]
```diff
@@ -1160,32 +1160,36 @@ <h4>Chords</h4>
...
+            </p>
+            <p>
+                Beam groups MUST NOT occur in Mensural encodings.
```
Comment (lpugin) [inline v2/index.html:1184]: Agree. Another think to check in our data.

File: v2/index.html (hunk 2 above)
Comment (ahankinson) [reply v2/index.html:1181]: It's a SHOULD NOT, not a MUST NOT, if that's any help.

File: v2/index.html (hunk 1 above)
Comment (ahankinson) [reply v2/index.html:1172]: Right

File: v2/index.html (hunk 2 above)
Comment (lpugin) [reply v2/index.html]: Yes, but why? This is quite common in music notation.

Comment (ahankinson) [reply v2/index.html]: I’ve updated the spec and removed that line

File: v2/index.html [hunk 4]
```diff
@@ -1208,6 +1205,19 @@ <h4>Beaming</h4>
                         <td class="notation-result"></td>
                         <td>Example beam encoding</td>
                     </tr>
+                    <tr class="notation-example">
+                        <td class="notation-code">
+                            <script type="application/json">
+                                {
+                                    "clef": "G-2",
+                                    "data": "8{ABqq{''CD}r'AB}"
```
Comment (lpugin) [inline v2/index.html:1213]:
```suggestion
                                    "data": "'8{ABqq{''6CD}r'8AB}"
```

File: v2/index.html [hunk 5]
```diff
@@ -1686,18 +1701,44 @@ <h5>Rhythmic sequence</h5>
...
+                change MAY appear anywhere in the music notation, but SHOULD appear immediately following a
+                bar line. This character MUST be followed by a time signature definition according to the
+                specifications given in the <a>Time Signatures</a> section.
             </p>
             <p>
-                The introductory characters are mandatory.
+                For any change of clef, key signature, time signature, or combination thereof, the change indication
+                MUST be separated from the notation that follows by a single space character. In the case of multiple
+                changes at once (for example, a key and a time signature change) the individual change indications
+                MUST NOT be separated by a space.
```
[the remark this answers is not in the document]
Comment (ahankinson) [reply v2/index.html:1733]: Sorry I have so many branches in flight I’m getting them mixed up

File: v2/index.html (hunk 3 above)
Comment (lpugin) [reply v2/index.html:1181]: Maybe it should be a `SHOULD`. It is not impossible to find some sources in proper mensural notation with *fusae* beamed together. I am thinking keyboard music from the 1600.

File: v2/index.html [hunk 6]
```diff
@@ -1160,32 +1160,33 @@ <h4>Chords</h4>
...
+                MUST NOT occur, except as part of a <a>grace note group</a>.
+            </p>
+            <p>
+                Notes with duration values of <code>0</code>, <code>1</code>, <code>2</code>,
+                <code>4</code>, and <code>9</code> MUST NOT occur within
+                a beam group. Notes with other duration values MAY occur within
+                a beam group.
+            </p>
+            <p>
+                Beam groups MUST NOT occur in Mensural encodings.
```
Comment (lpugin) [inline v2/index.html:1181]:
```suggestion
                Beam groups SHOULD NOT occur in Mensural encodings.
```

Comment (lpugin) [reply v2/index.html:1181]: We can always go back to it

**Comment by ahankinson** [comment]
I'm going to assume that this is now OK and merge it so I can clean up the many branches I have on the go, so that I can keep moving forward with the spec.

**Pull request merged**